### ISSUE TITLE
The Issues With the Current Chat System 

### ISSUE THREAD

**Issue by Blooshing** [ppy/osu-web#8473]
Hello! These issues were founded from: [https://github.com/ppy/osu/discussions/14830](url) 
 [https://github.com/ppy/osu/discussions/15213](url)

### Localized Pop-up for Chat Rules/Game Rules

<p>Hey all,

>I was wondering if there was a plan or an idea to implement a pop-up/ToS/something that would introduce or define the chat rules for the first time that a player were to open the chat, either the first time or on every first time that playing session, in their respective language that can be detected by the settings or however osu! usually does. If not then the language is easily changeable in the settings.

>Some of the main issues that I see in #osu (specifically):
>>New players speaking in their native tongue, either on accident, or then on purpose after the chat tries to direct them to the right channel. (They seem to not understand that #osu is english only)
Chat topics can be a disaster because not everyone knows that osu! is for all ages and some topics are just not suitable for the younger crowd.
General chat rules like: Spamming, flooding, creating a ladder game, don't be sexist, racist, homophobic, transphobic, (In general just being nice and a respectful human being).
>>>Other smaller issues about streaming, correct channels, correct language for the channel you are currently in, etc.
>>>>Another big issue that is more of an annoyance than anything is the constant flood of multiplayer lobbies being broadcasted into #osu. It has become such an issue that the only way that people would ever listen to this is if some how banchobot/the rule system of #osu would immediately remove the message and put it into #lobby or flat out remove the text and bancho sends the user the !faq multiplayer command or the !faq lobby faq.</p>

### Missing/Not added report feature in lazer's chat.

>Seems that in comparison to stable/anything not lazer there is something other than the !report , whether it be the right click of the mouse and either pressing 6 or clicking the report button and then filling out the reason(s) there for either simplicity or nuance's sake. I believe using the GUI is probably the most efficient and most known way of reporting.  
>>However, there is no such feature like that in lazer with the screenshots been taken 10/21/2021 and show no report feature if you are looking at the webpage version of the userpage or if there would be an in-client report feature (besides the command). 

>>>![143930253-a25f30b1-2665-4efd-8c48-e247727b60d7](https://user-images.githubusercontent.com/90729492/147530486-98eb0f62-9c20-4d3b-89e7-a6c8ac09b62e.jpg)
>>>![143930302-1db257a2-d657-4bcf-8d03-cb6d2b13f06f](https://user-images.githubusercontent.com/90729492/147530496-14f182b4-34c2-4d5e-8c12-0cae2d6fc6cc.png)

### Issue with the chat on the osu!web (discovered by wowcake): No word-filtering present: 

>At the current state, the chat in osu!web has no filtering, meaning you could send many offensive words without them being filtered, for example, saying the "fuck" word will stay as that on the user receiver's screen, instead of being filtered and sent as "firetruck" This could potentially lead into unnecessary annoyance and an abuse of this usage, which might attract other users' attention of how to do so, and resulting in those people finding their way to abuse it.

Hopefully this can get some traction as lazer's chat is somewhat incomplete even with the rework coming up (soon :tm:) 

Thanks for the read and sorry that it is posted in multiple directories, I was told to post it here as well. 

### Another issue I have come across since posting this:

>Seems that here is no #highlight in lazer's chat. The only option to see if someone notified you would be the notification bar/panel; however, this is kind of obsolete because there is no audio notification or visual notification other than a small little red number on top of the notification bar icon. Very easy to miss that a highlight or any substantial notification was present. 
>>I suggest bringing back the highlight color system where the user that says another users name becomes highlighted to that user, I think it is highly efficient and is a great way to organize chat. Can be very annoying to try and search for a single persons message amongst dozens of people talking in #osu with no guidance.

**Comment by ZeroPyrozen** [comment]
In-game and web chat rules pop-up would be good to warn user about the channel rules. For example, twitch has custom chat rules pop-up that appear when the viewer click the chat box.

![](https://pyrozen.s-ul.eu/9fTbfTmh)



> Hello! These issues were founded from: https://github.com/ppy/osu/discussions/14830
https://github.com/ppy/osu/discussions/15213

Use `#14830` and `#15213` to mention issue/pr

**Comment by peppy** [comment]
Going to close this issue as most of the issues are already being tracked in the osu! repository (not sure why it was opened here) and the formatting of the opening post is very hard to follow.

The chat system is currently getting a lot of focus and things like "filtering" will be added as time progresses.
